### PR TITLE
py-shacl: new version, update dependencies

### PR DESCRIPTION
**Pull request by matz-e**
Also updates the dependencies py-prettytable and py-rdflib.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the new version sha256s.

File: var/spack/repos/builtin/packages/py-pyshacl/package.py [hunk 1]
```diff
@@ -13,18 +13,23 @@ class PyPyshacl(PythonPackage):
 
     license("Apache-2.0")
 
+    version("0.25.0", sha256="91e87ed04ccb29aa47abfcf8a3e172d35a8831fce23a011cfbf35534ce4c940b")
     version("0.20.0", sha256="47f014c52cc69167b902c89b3940dd400f7f5d2169a62f97f837f3419b4a737d")
     version("0.17.2", sha256="46f31c7a7f7298aa5b483d92dbc850ff79a144d26f1f41e83267ed84b4d6ae23")
 
     depends_on("py-poetry-core@1.1:1", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyshacl/package.py:20]: `1.8.1:1` is required now

File: var/spack/repos/builtin/packages/py-pyshacl/package.py [hunk 2]
```diff
@@ -13,18 +13,23 @@ class PyPyshacl(PythonPackage):
...
     depends_on("python@3.7:3", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyshacl/package.py:22]: `3.8.1:3` is required now

File: var/spack/repos/builtin/packages/py-pyshacl/package.py [hunk 3]
```diff
@@ -13,18 +13,23 @@ class PyPyshacl(PythonPackage):
...
     depends_on("py-rdflib@6.0.0:6", when="@0.17.2", type=("build", "run"))
-    depends_on("py-rdflib@6.2.0:6", when="@0.20.0:", type=("build", "run"))
+    depends_on("py-rdflib@6.2.0:6", when="@0.20.0", type=("build", "run"))
+    depends_on("py-rdflib@6.3.2:7", when="@0.25.0:", type=("build", "run"))
     depends_on("py-html5lib@1.1:1", when="@0.20.0:", type=("build", "run"))
+    depends_on("py-importlib-metadata@6:", when="@0.25.0: ^python@:3.11", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyshacl/package.py]:
```suggestion
    depends_on("py-importlib-metadata@7:", when="@0.25.0: ^python@:3.11", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pyshacl/package.py [hunk 4]
```diff
@@ -13,18 +13,23 @@ class PyPyshacl(PythonPackage):
...
     depends_on("py-owlrl@5.2.3:6", when="@0.17.2", type=("build", "run"))
     depends_on("py-owlrl@6.0.2:6", when="@0.20.0:", type=("build", "run"))
     depends_on("py-packaging@21.3:", when="@0.20.0:", type=("build", "run"))
-    depends_on("py-prettytable@2.2.1:2", type=("build", "run"))
+    depends_on("py-prettytable@2.2.1:2", when="@:0.20", type=("build", "run"))
+    depends_on("py-prettytable@3.5.0:", when="@0.25.0: ^python@:3.11", type=("build", "run"))
+    depends_on("py-prettytable@3.7.0:", when="@0.25.0: ^python@3.12:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyshacl/package.py:34]: I like to sort dependencies in the same order as `pyproject.toml`, with newer versions first. This makes it easier to review PRs. Not required though, just a style thing.

File: var/spack/repos/builtin/packages/py-rdflib/package.py [hunk 5]
```diff
@@ -23,6 +23,7 @@ class PyRdflib(PythonPackage):
 
     license("BSD-3-Clause")
 
+    version("7.0.0", sha256="9995eb8569428059b8c1affd26b25eac510d64f5043d9ce8c84e0d0036e995ae")
     version("6.3.2", sha256="72af591ff704f4caacea7ecc0c5a9056b8553e0489dd4f35a9bc52dbd41522e0")
     version("6.2.0", sha256="62dc3c86d1712db0f55785baf8047f63731fa59b2682be03219cb89262065942")
     version("6.0.2", sha256="6136ae056001474ee2aff5fc5b956e62a11c3a9c66bb0f3d9c0aaa5fbb56854e")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rdflib/package.py:29]: Python `3.8.1:3` required now

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyshacl/package.py (hunk 4 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-pyshacl/package.py:34]: Thanks, I will keep that in mind for future PRs!

I've updated the other requirements as reviewed, sorry that took a little bit.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**